### DISCUSSION TITLE
トラックボールが挙動しない（Pro Micro 周辺が安定しない）

### DISCUSSION THREAD

**Discussion by maonyoro** [Yowkees/keyball#354] [Q&A]
お世話になっております。先月くらいから Keyball 61 を利用させていただいております。
ここ数週間、どうしても解決できない事象があり質問させてください 🙇‍♀️ 

類似質問は全て目を通しております。

### 事象
- トラックボールが動作しない
  - 元々動作していたが、片方端子がもげていた Pro Micro の換装を試みた後から動作不良
- Pro Micro 挙動が安定しない
  - 切り分けのため 4~5 個、手元に Pro Micro があるが、どれも安定動作しない
    - OLEDモジュール 表示されない、一部キーの反応がない、など
      - 全キー動作する個体もあるので、単純に半田付け等の不良かもしれないのですが。。

### 確認したこと
![右手ウラ側](https://github.com/Yowkees/keyball/assets/8395897/d2001c45-fcba-4020-b161-eaa1132ea627)

### 考えていること
裏側はうまくいってそうなので、
- [1] Pro Micro → 裏側、に至る箇所 ＝ 12ピンソケット
- [2] トラックボール読取り基盤
のどちらかかな、、と思っています。。

Pro Micro の動作が（４つ５つ試しても）不安定、というところで、ピンソケットを疑っています。
確かに何度も Pro Micro の半田付けの土台になったり、裏を何度か半田しなおしたりしたので心当たりも若干あります。

この内容で、もう少し確認・切り分けできそうな箇所などはありますでしょうか…。
類似質問が多い中、大変恐縮ですがご確認いただければ幸いです。よろしくお願いいたします。

**Comment by maonyoro** [comment]
補足情報になりそうな写真を添付いたします 🙇‍♀️ 

## Pro Micro 裏側

![promicro_ura1](https://github.com/Yowkees/keyball/assets/8395897/76da2b32-b77b-4f72-8670-f4e1a95d3291)
![promicro_ura2](https://github.com/Yowkees/keyball/assets/8395897/425e3214-2dae-406e-b625-20d1736b1620)

<br>

## トラックボール裏

![ball_ura1](https://github.com/Yowkees/keyball/assets/8395897/1f9f3e9c-1711-4030-9a93-b5a0c00fd7a6)

<br>

## トラックボール読み取り基盤

![IMG_3736](https://github.com/Yowkees/keyball/assets/8395897/4a897531-a9d7-4971-b61b-69c23b5393cc)

<br>

## Pro Micro

### トラックボール以外動作する個体

![IMG_3739](https://github.com/Yowkees/keyball/assets/8395897/22331d5c-5ebf-42e2-a9b8-5e47ce525c5e)

### トラックボールに加えて部分的なキー不良や OLED モジュール表示されない等

![IMG_3738](https://github.com/Yowkees/keyball/assets/8395897/f8478d27-8f20-4df5-8088-6dafafb5528f)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

トラックボールの不調については、半田付けには特に悪そうなところは見られませんでしたし、導通も確認されているということで、L字7Pコンスルーの不具合を確認したいと思います。
下記画像のようにピンの変形がないかご確認ください。画像くらいの変形でも導通不良が発生していた事例になります。

![2_conthru_henkei](https://github.com/Yowkees/keyball/assets/84209772/afa4fb94-c9e7-4eeb-9b26-04646e959375)

解決しない場合は、トラックボール基板単体での販売も可能ですので、白銀ラボウェブショップのお問合せよりご連絡ください。

ProMicroによりキー不具合やOLED不具合がある件についてはやはりProMicroの半田付けを見直すしかないかと思います。
ProMicroへ取り付ける12Pピンヘッダーはキットに同梱しているもの（PSS-410153-12という製品です）を使用したほうが、ピンソケットとの信頼度があがると思いますのでご検討ください。

以上、お手数ですがご対応よろしくお願いいたします。

**Comment by maonyoro** [comment]
お早いお返事いただき、どうもありがとうございます。
7P コンスルーについても確認してみましたが、問題なさそうに見えました。

![IMG_3740](https://github.com/Yowkees/keyball/assets/8395897/463c15e5-0698-463d-95b5-b565991213d3)
![IMG_3741](https://github.com/Yowkees/keyball/assets/8395897/e9b54ee5-c1a2-4bfb-8bcd-c997c080d47f)

<br>

---

やはり不安なパーツの交換・切り分けをするしかなさそうな気がするので、改めて白銀ラボさんのお問い合わせよりご連絡させていただこうと思います。どうもありがとうございます。
ちなみに、7P コンスルーの端子に対して、何か擬似的？に電気的に動作確認する術、ってあるものでしょうか…？
（キーボード基盤 or トラックボール基盤、の問題切り分けができないものかな、と）

ProMicro 周りは、一部試行の際に ProMicro 付属ピンヘッダを使いつつも、基本はキットのピンヘッダを利用しておりまして…
直近でも白銀ラボさんからピンヘッダだけ購入させていただいたのですが、こんなに失敗するかな？？というくらいうまくいってなかったので今回の質問をするに至った次第です。。
問題なく動作するものがある時点で半田周りなんだろうな…と思いつつも、もはや半田を増やすべきなのか減らすべきなのかすらわからず半泣き状態になっていました…苦笑

**Comment by maonyoro** [comment]
お世話になっております。

結果的にピンソケットの不良だったようで、新しいピンソケットに付け替えたことで全て問題なくどうさするようになりました。
基盤側も ProMicro の半田付け等も問題なかったようですが、基盤と ProMicro の接点となるピンソケットが一部導通不良だったようでした。

大変勉強になりました。また今日から愛用させていただこうと思います。
本コメントにてクローズしておきます。ありがとうございました。
